File: JavaScript/704_Binary_Search/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
const search = function (nums, target) {
    let leftPtr = 0;
    let rightPtr = nums.length - 1;

    while (leftPtr <= rightPtr) {
        const midPtr = Math.floor(leftPtr / 2 + rightPtr / 2);

        if (nums[midPtr] == target) {
            return midPtr;
        } else if (nums[midPtr] < target) {
            leftPtr = midPtr + 1;
        } else {
            rightPtr = midPtr - 1;
        }
    }

    return -1;
};

File: Java/704_Binary_Search/Solution.java
class Solution {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;

        while (left <= right) {
            final int mid = (left + right) / 2;
            final int midNum = nums[mid];

            if (midNum == target) {
                return mid;
            } else if (midNum < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        return -1;
    }
}